pandas/core/ops.py

```diff
@@ -1355,8 +1355,9 @@ def wrapper(self, other, axis=None):
                                      name=res_name)
 
         elif (is_extension_array_dtype(self) or
-              (is_extension_array_dtype(other) and
-               not is_scalar(other))):
+              (is_extension_array_dtype(other) and not is_scalar(other))):
+            # Note: the `not is_scalar(other)` condition rules out
+            # e.g. other == "category"
```

Comment:
the issue here is we need to be able to distinguish between an actual dtype comparison and a real comparison, e.g.
``df.dtypes == 'category' `` (or ``df.dtypes == 'Int8' ``.

this is pretty thorny, e.g. how do you know when to convert a scalar string in a comparison op to an actual dtype for comparisons

Reply:
Yah.  A while ago there was a check `is_categorical_dtype(y) and not is_scalar(y)` and it took me a while to figure out that the `is_scalar` part was specifically to avoid letting "category" through, so I've gotten in the habit of adding this comment for future readers.